CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change. 
+
+Please note we have a [Code of Conduct](CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
+
+## <a name="issue"></a> Found an Issue?
+If you find a bug in the source code or a mistake in the documentation, you can help us by
+[submitting an issue](#submit-issue) to the GitHub Repository. Even better, you can
+[submit a Pull Request](#submit-pr) with a fix.
+
+## <a name="feature"></a> Want a Feature?
+You can *request* a new feature by [submitting an issue](#submit-issue) to the GitHub
+Repository. If you would like to *implement* a new feature, please submit an issue with
+a proposal for your work first, to be sure that we can use it.
+
+* **Small Features** can be crafted and directly [submitted as a Pull Request](#submit-pr).
+
+## <a name="submit"></a> Contribution Guidelines
+
+### <a name="submit-issue"></a> Submitting an Issue
```

Comment:
Why not `<a name="submit-issue"/>`?.. would be simpler and easier to read since it's just an anchor.

Reply:
Also, AFAIK, markdown support auto-generating anchors for section header... at least IDEA has a auto-suggestion for those... I think they are more readable (something like `#submitting-an-Issue) in this case. What is the advantage of explicit `<a/>` anchors?

Reply:
@csplinter ^

Reply:
Really they both render appropriately in github which is what really matters.

Reply:
I don't have a preference